Assets/Scripts/Entity/Player/Sanity/SanityPostProcessing.cs

```diff
@@ -0,0 +1,181 @@
+        [SerializeField]
+        private int contrast;
+
+        [Tooltip("The higher the shutter angle, the more motion blurr.")]
+        [SerializeField]
+        private int maxShutterAngle;
+
+        [Tooltip("Frequency of vignette pulsing.")]
+        [SerializeField]
+        private float vignettePulseFrequency;
+
+        [Tooltip("Intensity of the vignette pulse.")]
+        [SerializeField]
+        [Range(0, 1)]
+        private float vignettePulseIntensity;
+
+        [Tooltip("Frequency of chromatic aberration pulsing.")]
+        [SerializeField]
+        private float aberrationPulseFrequency;
+
+        [Tooltip("Intensity of chromatic aberration pulse.")]
+        [SerializeField]
+        [Range(0, 1)]
+        private float aberrationPulseIntensity;
+
+
+        [Range(0,100)]
+        private int _sanity;
+        // have different frequencies for different effects
+        private bool[] _isGrowing;
+
+        //filters:
+        private PostProcessVolume _postProcessVolume;
+
+        private ColorGrading _colorGrading;
+        private int _currentSaturation;
+        private int _currentContrast;
+
+        private MotionBlur _motionBlur;
+        private int _baseShutterAngle;
+        private int _currentShutterAngle;
+
+        private Vignette _vignette;
+        [Range(0, 1)]
+        private float _vignetteBaseIntensity;
+        [Range(0, 1)]
+        private float _currentVignetteIntensity;
+        private float _vignettePulse;
+
+        private ChromaticAberration _chromaticAberration;
+        [Range(0, 1)]
+        private float _aberrationBaseIntensity;
+        [Range(0, 1)]
+        private float _currentAberrationIntensity;
+        private float _aberrationPulse;
+
+        
+        private void Awake()
+        {
+            _isGrowing = new bool[3];
+            _postProcessVolume = GetComponent<PostProcessVolume>();
+            _postProcessVolume.profile.TryGetSettings(out _colorGrading);
+            _postProcessVolume.profile.TryGetSettings(out _motionBlur);
+            _postProcessVolume.profile.TryGetSettings(out _chromaticAberration);
+            _postProcessVolume.profile.TryGetSettings(out _vignette);
+            _sanity = 100; // init with 100, since the first sent out event doesn't seem to be received here
+        }
+
+        /// <summary>
+        /// Remembers base intensities (resp. shutter angle).
+        /// </summary>
+        private void Start()
+        {
+            _aberrationBaseIntensity = _chromaticAberration.intensity.value;
+            _vignetteBaseIntensity = _vignette.intensity.value;
+            _baseShutterAngle = (int) _motionBlur.shutterAngle.value;
+            ClampAll();
+        }
+
+        private void OnEnable()
+        {
+            PlayerState.OnPlayerSanityUpdate += (int sanity) => _sanity = sanity;
+        }
+
+        private void OnDisable()
+        {
+            PlayerState.OnPlayerSanityUpdate -= (int sanity) => _sanity = sanity;
```

Comment:
This is a lambda function and will not remove the added function. Write a method for it and (un-)subscribe with that one

Reply:
haha okay I knew it 😂